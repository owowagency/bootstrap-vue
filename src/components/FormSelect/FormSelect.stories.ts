import FormSelect from '.';
import {sizes} from '@/composables/useSize';

export default {
    title: 'Forms/Form select',
    component: FormSelect,
    argTypes: {},
};

const defaultArgs = {
    items: [
        {value: 'Item 1'},
        {value: 'Item 2', label: 'ITEM 2', selected: true},
        {value: 'Item 3'},
    ],
};

const template = (args) => ({
    components: {FormSelect},
    setup: () => ({args}),
    template: '<FormSelect v-bind="args" />',
});

export const Default = template.bind({});

Default.args = defaultArgs;

export const Disabled = template.bind({});

Disabled.args = {
    ...defaultArgs,
    disabled: true,
};

export const Sizing = (args) => ({
    components: {FormSelect},
    setup: () => ({args, sizes}),
    template: '<FormSelect v-for="s of sizes" class="mb-1" :size="s" v-bind="args" />',
});

Sizing.args = defaultArgs;
