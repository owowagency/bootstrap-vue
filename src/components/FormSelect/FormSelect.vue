<template>
    <select
        class="form-select"
        :class="classes"
        :disabled="disabled"
    >
        <option
            v-for="item in items"
            :key="item.value"
            :selected="item.selected"
            :value="item.value"
        >
            {{ item.label || item.value }}
        </option>
    </select>
</template>
<script lang="ts">
import {formSelectProps} from '@/composables/useFormSelect';
</script>

<script lang="ts" setup>
import useFormSelect from '@/composables/useFormSelect';

const props = defineProps(formSelectProps);

const {classes} = useFormSelect(props);
</script>

<docs>
```vue
<FormSelect
    v-model="language"
    :items="[Item]"
/>
```

## Imported Props

| Prop name | Description                    | Type                                         | Values | Default  |
| --------- | ------------------------------ | -------------------------------------------- | ------ | -------- |
| disabled  | Disables the component         | boolean                                      | -      | false    |
| items     | The items of the dropdown menu | [Item[]](../../composables/useDropdownItems) | -      | () => [] |
| size      | The size of the component      | [Size](../../composables/useSize) (string)   | -      | 'md'     |
</docs>
