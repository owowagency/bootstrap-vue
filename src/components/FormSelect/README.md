# FormSelect

```vue
<FormSelect v-model="language" :items="[Item]" />
```

## Imported Props

| Prop name | Description                    | Type                                         | Values | Default  |
| --------- | ------------------------------ | -------------------------------------------- | ------ | -------- |
| disabled  | Disables the component         | boolean                                      | -      | false    |
| items     | The items of the dropdown menu | [Item[]](../../composables/useDropdownItems) | -      | () => [] |
| size      | The size of the component      | [Size](../../composables/useSize) (string)   | -      | 'md'     |
