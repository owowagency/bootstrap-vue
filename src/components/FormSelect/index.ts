import FormSelect from './FormSelect.vue';

export default FormSelect;
