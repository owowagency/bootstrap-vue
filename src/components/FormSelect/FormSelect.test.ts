import {describe, expect, it} from 'vitest';
import FormSelect from '.';
import {shallowMount} from '@vue/test-utils';

describe('template', () => {
    componentRenderTest(FormSelect, {
        props: {
            items: [
                {value: 'Item 1'},
                {selected: true, value: 'Item 2'},
                {label: 'Item three', value: 'Item 3'},
            ],
        },
    });

    componentWrapperClassTest(FormSelect, {disabled: true}, 'disabled');

    componentWrapperClassTest(FormSelect, {size: 'lg'}, 'form-select-lg');

    it('adds disabled property', () => {
        const wrapper = shallowMount(FormSelect, {
            props: {disabled: true},
        });

        expect((wrapper.element as HTMLSelectElement).disabled).toBe(true);
    });
});
