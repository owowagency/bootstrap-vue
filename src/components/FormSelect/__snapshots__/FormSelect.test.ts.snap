// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<select class=\\"form-select form-select-md\\">
  <option value=\\"Item 1\\">Item 1</option>
  <option value=\\"Item 2\\">Item 2</option>
  <option value=\\"Item 3\\">Item three</option>
</select>"
`;
